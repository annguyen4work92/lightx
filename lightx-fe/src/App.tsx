import ListUserPage from 'pages/list-user'
import React from 'react'
import './App.css'

const App = (): React.ReactElement => {
  return (
    <div className="App">
      <ListUserPage />
    </div>
  )
}

export default App
