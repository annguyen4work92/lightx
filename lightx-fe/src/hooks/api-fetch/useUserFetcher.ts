import { EndPoint, getApiUrl } from 'configs'
import React from 'react'
import { useHttpProvider } from 'hooks/base'

export interface IUserType {
  userId: number
  id: number
  title: string
  completed: boolean
}

export const useUserFetcher = ()=> {
  const [dataSource, setDataSource] = React.useState<IUserType[]>([])
  const { httpProvider } = useHttpProvider()

  const fetchUser = async () => {
    const res = await httpProvider.get<IUserType[]>(
      getApiUrl(EndPoint.GetList)
    )
    setDataSource(res.data as IUserType[])
    return res.data
  }

  React.useEffect(() => {
    fetchUser()
  }, [])

  return {
    dataSource,
    fetchUser
  }
}
