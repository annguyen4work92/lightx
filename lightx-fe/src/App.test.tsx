import { cleanup, render, renderHook, screen, waitFor } from '@testing-library/react'
import { useUserFetcher } from 'hooks/api-fetch/useUserFetcher'
import ListUserPage from 'pages/list-user'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import App from './App'

afterAll(cleanup);
// beforeEach(() => console.log('Start a test'));

function sleep(ms: number) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

test('renders learn react link', () => {
  act(() => {
    render(<App />)
  })
  const linkElement = screen.getByText(/Here is the header/i)
  expect(linkElement).toBeInTheDocument()

  const id = screen.getByTestId('testid'); // data-testid
  expect(id).toBeInTheDocument()
  expect(id).toBeEnabled();
  const disabledInput = screen.getByTestId('disabled-input'); // data-testid
  expect(disabledInput).toBeInTheDocument()
  expect(disabledInput).toBeInTheDocument()
  expect(disabledInput).toBeDisabled(); // Check item is enabled.
})

test('App snapshot', async () => {
  const appRenderer = await act(() => renderer.create(<ListUserPage />));
  const { result } = renderHook(() => useUserFetcher());
  await waitFor(async () => {
    await sleep(3000);
    expect(appRenderer.toJSON()).toMatchSnapshot()
    expect(result.current.dataSource).toMatchSnapshot();
  }, {timeout: 4000})
})

test('renders learn react Button', () => {
  act(() => {
    render(<App />)
  })
  const ButtonElement = screen.getByText(/Hehes/i)
  expect(ButtonElement).toBeInTheDocument()
})

test('renders data', async () => {
  act(() => {
    render(<ListUserPage />)
  })
  await waitFor(() => {
    const ButtonElement = screen.getByText(/"userId":/i)
    expect(ButtonElement).toBeInTheDocument()
  })
})
