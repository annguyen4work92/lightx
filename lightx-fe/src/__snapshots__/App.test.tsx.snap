// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="layout-wrapper"
>
  <div
    className="header-wrapper"
  >
    Here is the header overrided
  </div>
  <div
    className="sidbar-wrapper"
  >
    Bo may la sidebar
  </div>
  <div
    className="content-wrapper"
  >
    <button>
      Hehes
    </button>
    <input
      data-testid="testid"
      id="hihi"
    />
    <input
      data-testid="disabled-input"
      disabled={true}
      id="disabled-input"
    />
    {"userId":1,"id":1,"title":"delectus aut autem","completed":false}
  </div>
  <div
    className="footer-wrapper"
  >
    I am footer
  </div>
</div>
`;

exports[`App snapshot 2`] = `
Object {
  "completed": false,
  "id": 1,
  "title": "delectus aut autem",
  "userId": 1,
}
`;
